src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php

```diff
@@ -524,4 +534,96 @@ protected function getObject()
 
         return $obj;
     }
+
+    /**
+     * @test
+     */
+    public function testEncodeXmlWithBoolValue()
+    {
+        $expectedXml = '<?xml version="1.0"?>
+<response><foo>1</foo><bar>0</bar></response>
+';
+
+        $actualXml = $this->encoder->encode(array('foo' => true, 'bar' => false), 'xml');
```

Comment:
I don't get why this is expected? IMO it should be `array('foo' => '1', 'bar' => '0')` unless explicitly defined as bool in a XML schema.